### PR TITLE
[pbconvert] Add protobuf convertor for sm

### PR DESCRIPTION
**Pull request by mlohvynenko**

### CODE REVIEW

File: include/pbconvert/common.hpp [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+::common::v1::ErrorInfo ConvertAosErrorToProto(const Error& error);
+
+/**
+ * Converts aos instance ident to protobuf.
+ *
+ * @param src instance ident to convert.
+ * @param[out] dst protobuf instance ident.
+ * @return void.
+ */
+void ConvertToProto(const InstanceIdent& src, ::common::v1::InstanceIdent& dst);
```
Comment (al1img) [inline include/pbconvert/common.hpp:34]: All API returns value. Make this one to match common pattern.

File: include/pbconvert/sm.hpp [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+
+namespace aos::common::pbconvert::sm {
+/**
+ * Converts aos push log to protobuf.
+ *
+ * @param src push log to convert.
+ * @param[out] dst protobuf log data.
+ * @return void.
+ */
+void ConvertToProto(const cloudprotocol::PushLog& src, ::servicemanager::v4::LogData& dst);
```
Comment (al1img) [inline include/pbconvert/sm.hpp:27]: Here as well some API, returns value through param ref some by return. Could we have same pattern for all? Either return all in return or all by ref. Also mixed pattern is possible for example use return for basic types and ref for struct classes etc.

File: include/pbconvert/sm.hpp [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+#include <aos/common/cloudprotocol/alerts.hpp>
+#include <aos/common/cloudprotocol/envvars.hpp>
+#include <aos/common/cloudprotocol/log.hpp>
+#include <aos/common/monitoring/monitoring.hpp>
+#include <aos/common/types.hpp>
+#include <aos/sm/networkmanager.hpp>
+
+#include <servicemanager/v4/servicemanager.grpc.pb.h>
+
+namespace aos::common::pbconvert::sm {
```
Comment (al1img) [inline include/pbconvert/sm.hpp:19]: remove ::sm namespace here. Otherwise looks not consistent.

File: src/iamclient/publicservicehandler.cpp [hunk 4]
```diff
@@ -87,7 +87,9 @@ void PublicServiceHandler::Close()
 
     LOG_INF() << "Closing public service handler";
 
-    for (auto& [_, subscription] : mSubscriptions) {
+    for (auto& keyValue : mSubscriptions) {
```
Comment (al1img) [inline src/iamclient/publicservicehandler.cpp:90]: Looks like this is normal construction. Better to suppress the warning.

**Review by al1img — APPROVED** [review]

**Review by MykolaSolyanko — APPROVED** [review]

**Review by mykola-kobets-epam — APPROVED** [review]

**Pull request merged**